sysl2/sysl/ints.go

```diff
@@ -0,0 +1,116 @@
+package main
+
+import (
+	"flag"
+	"io"
+	"os"
+	"regexp"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+func GenerateIntegrations(
+	root_model, title, output, project, filter, modules string,
+	exclude []string, clustered, epa bool,
+) map[string]string {
+	r := make(map[string]string)
+	mod := loadApp(root_model, modules)
+
+	if len(exclude) == 0 && project != "" {
+		exclude = append(exclude, project)
+	}
+	excludeStrSet := MakeStrSet(exclude...)
+	ds := NewDependencySet()
+	ds.CollectAppDependencies(mod)
+
+	// The "project" app that specifies the required view of the integration
+	app := mod.GetApps()[project]
+	of := MakeFormatParser(output)
+	// Interate over each endpoint within the selected project
+	for epname, endpt := range app.GetEndpoints() {
+		// build the set of excluded items
+		excludes := MakeStrSetFromSpecificAttr("exclude", endpt.GetAttrs())
+		passthroughs := MakeStrSetFromSpecificAttr("passthrough", endpt.GetAttrs())
+		// endpt.stmt's "action" will conatain the "apps" whose integration is to be drawn
+		// each one of these will be placed into the "integration" list
+		integrations := MakeStrSetFromActionStatement(endpt.GetStmt())
+
+		arr := ds.ToSlice()
```

Comment:
iterating call list again to convert to a slice, that too inside the loop!

Reply:
It's not iteration CALL LIST here. however, I moved it out of the loop.

Reply:
how are you converting the map to a slice?
Are you not iterating over the keys of the map and add the keys to the slice?
So effectively iterating over the map again.

Reply:
finally remove the traversing for ds.Deps, we don't care about the sorting for Deps.